pkg/loki/loki.go

```diff
@@ -111,15 +111,20 @@ func (t *Loki) setupAuthMiddleware() {
 }
 
 func (t *Loki) init(m moduleName) error {
-	if _, ok := t.inited[m]; ok {
-		return nil
-	}
-
-	for _, dep := range modules[m].deps {
-		if err := t.init(dep); err != nil {
+	// initialize all of our dependencies first
+	for _, dep := range getDeps(m) {
+		if err := t.initModule(dep); err != nil {
 			return err
 		}
 	}
+	// lastly, initialize the requested module
+	return t.initModule(m)
+}
+
+func (t *Loki) initModule(m moduleName) error {
+	if _, ok := t.inited[m]; ok {
+		return nil
+	}
```

Comment:
What purpose does this serve now?  It used to exist to ensure we only stated/stopped modules once, but it that now guaranteed by `getDeps`?